contracts/Registrar.sol

```diff
@@ -15,8 +15,8 @@ contract Registrar {
     /// The Rad/Eth exchange.
     address public exchangeAddress;
 
-    /// The Rad ERC20 token.
-    address public immutable radAddress;
+    /// The Radicle ERC20 token.
+    ERC20Burnable public immutable rad;
```

Comment:
That one is hairy, RadicleToken isn't really `ERC20Burnable`, it lacks many non-ERC20 functions present in OpenZeppelin's implementation. It will work thanks to duck typing of the dynamic dispatch and because we simply don't use the unimplemented functions, but it's VERY ugly and misleading. The best solution would be to create `IERC20Burnable` deriving from `IERC20`.

BTW I've noticed that RadicleToken IS NOT an ERC-20 token because e.g. it exposes `DECIMALS()` instead of `decimals()`. I haven't fixed it yet, but we absolutely have to straighten it before any release. Having a proper interface would catch that.

Reply:
For now, since we're using `Rad.sol`, it's okay. We can change it when we migrate to `RadicleToken.sol`, and just have it as `RadicleToken`.

`RadicleToken` definitely is ERC20 compatible, I think `decimals()` is optional. We should however maybe enforce it via the interface.

Reply:
That's fair, for now let's keep it in mind.

> RadicleToken definitely is ERC20 compatible, I think decimals() is optional. We should however maybe enforce it via the interface.

Nice, I wasn't aware that parts of ERC-20 are optional. But we're still missing `totalSupply` :P

Reply:
Yeah just noticed :joy: 

We should probably implement it, although it doesn't seem to pose a problem for COMP, I'm guessing because wallets don't actually ever call that function..